### PR TITLE
Add nxos_bgp_neighbor_address_family resource module

### LINKED ISSUE #155: BGP Modules support  peer-policy
##### SUMMARY
Request that the BGP process and neighbor modules support peer-polocies 

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
cisco.nxos.nxos_bgp_af
cisco.nxos.nxos_bgp
cisco.nxos.nxos_bgp_neighbor_af

##### ADDITIONAL INFORMATION
The goal of this feature is better to manage BGP neighbors by taking the load of the cpu/ram of the device 

NXOS code 

router bgp 65200
  router-id 1.1.1.1
  log-neighbor-changes
  address-family ipv4 unicast
  template peer-policy OUTBOUND_ROUTES
    route-map TO_OUTBOUND_ROUTES out
    soft-reconfiguration inbound always
  neighbor 10.67.252.14 remote-as 64570
    description PEER
    address-family ipv4 unicast
      inherit peer-policy OUTBOUND_ROUTES 10
      route-map PEER  in

### PR DESCRIPTION
**Pull request by NilashishC**
Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

##### SUMMARY
- Targets https://github.com/ansible-network/resource_module_models/pull/115
- Add nxos_bgp_neighbor_address_family resource module.
- Fixes #155 
##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
nxos_bgp_neighbor_address_family

### CODE REVIEW

**Comment by NilashishC** [comment]
Module | coverage
-- | -- |
plugins/module_utils/network/nxos/argspec/bgp_neighbor_address_family/__init__.py | 100%
plugins/module_utils/network/nxos/argspec/bgp_neighbor_address_family/bgp_neighbor_address_family.py | 100%
plugins/module_utils/network/nxos/config/bgp_neighbor_address_family/__init__.py | 100%
plugins/module_utils/network/nxos/config/bgp_neighbor_address_family/bgp_neighbor_address_family.py | 98%
plugins/module_utils/network/nxos/facts/bgp_neighbor_address_family/__init__.py | 100%
plugins/module_utils/network/nxos/facts/bgp_neighbor_address_family/bgp_neighbor_address_family.py | 99%
plugins/module_utils/network/nxos/rm_templates/bgp_neighbor_address_family.py | 96%
plugins/modules/nxos_bgp_neighbor_address_family.py | 88%
tests/unit/modules/network/nxos/test_nxos_bgp_neighbor_address_family.py | 98%
Total | 98%

File: plugins/module_utils/network/nxos/config/bgp_neighbor_address_family/bgp_neighbor_address_family.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+            self._compare(want={}, have=h_vrf, vrf=k)
+
+    def _bgp_list_to_dict(self, data):
+        if "neighbors" in data:
+            for nbr in data["neighbors"]:
+                if "address_family" in nbr:
+                    nbr["address_family"] = {
+                        (x["afi"], x.get("safi")): x
+                        for x in nbr["address_family"]
+                    }
```
Comment (GomathiselviS) [inline plugins/module_utils/network/nxos/config/bgp_neighbor_address_family/bgp_neighbor_address_family.py:204]: The dict comprehensions used in this function may result in sanity failures.

File: plugins/module_utils/network/nxos/rm_templates/bgp_neighbor_address_family.py [hunk 2]
```diff
@@ -0,0 +1,877 @@
...
+        cmd += " {generate_warning_threshold}".format(**data)
+    if "restart_interval" in data:
+        cmd += " restart {restart_interval}".format(**data)
+    if data.get("warning_only"):
+        cmd += " warning-only"
+    return cmd
+
+
+def _tmplt_next_hop_third_party(data):
+    pass
```
Comment (GomathiselviS) [inline plugins/module_utils/network/nxos/rm_templates/bgp_neighbor_address_family.py:36]: Is this function definition needed?

**Review by GomathiselviS — CHANGES_REQUESTED** [review]
Few nitpicks !

File: plugins/module_utils/network/nxos/rm_templates/bgp_neighbor_address_family.py (hunk 2 above)
Comment (NilashishC) [reply plugins/module_utils/network/nxos/rm_templates/bgp_neighbor_address_family.py:36]: Oh, yes. This is redundant.

File: plugins/module_utils/network/nxos/config/bgp_neighbor_address_family/bgp_neighbor_address_family.py (hunk 1 above)
Comment (NilashishC) [reply plugins/module_utils/network/nxos/config/bgp_neighbor_address_family/bgp_neighbor_address_family.py:204]: The dict/list comprehensions are only invalid for Python 2.6 tests which we don't support and are skipped.

**Review by ashwini-mhatre — APPROVED** [review]

**Review by GomathiselviS — APPROVED** [review]

**Pull request merged**